mteb/tasks/Retrieval/code/AppsRetrieval.py

```diff
@@ -0,0 +1,110 @@
+    split = _EVAL_SPLIT
+
+    corpus = {split: {}}
+    queries = {split: {}}
+    relevant_docs = {split: {}}
+
+    corpus_identifier = "-queries-corpus"
+    dataset = datasets.load_dataset(
+        path + corpus_identifier,
+        cache_dir=cache_dir,
+        revision=revision,
+        trust_remote_code=True,
+    ).filter(lambda example: example["partition"] == "test")
+
+    corpus_data = dataset["corpus"]
+    for row in corpus_data:
+        doc_id = row["_id"]
+        doc_title = row["title"]
+        doc_text = row["text"]
+        corpus[split][doc_id] = {"title": doc_title, "text": doc_text}
+
+    # Load queries data
+    queries_data = dataset["queries"]
+
+    for row in queries_data:
+        query_id = row["_id"]
+        query_text = row["text"]
+        queries[split][query_id] = query_text
+
+    # Load relevant documents data
+    qrels_identifier = "-qrels"
+    qrels_data = datasets.load_dataset(
+        path + qrels_identifier,
+        cache_dir=cache_dir,
+        revision=revision,
+        trust_remote_code=True,
+    )
+
+    for row in qrels_data[split]:
+        query_id = row["query_id"]
+        doc_id = row["corpus_id"]
+        score = row["score"]
+        if query_id not in relevant_docs[split]:
+            relevant_docs[split][query_id] = {}
+        relevant_docs[split][query_id][doc_id] = score
+
+    corpus = datasets.DatasetDict(corpus)
+    queries = datasets.DatasetDict(queries)
+    relevant_docs = datasets.DatasetDict(relevant_docs)
+
+    return corpus, queries, relevant_docs
+
+
+class AppsRetrieval(AbsTaskRetrieval):
+    metadata = TaskMetadata(
+        name="AppsRetrieval",
+        description="The dataset is a collection of natural language queries and their corresponding code snippets. The task is to retrieve the most relevant code snippet for a given query.",
+        reference="https://arxiv.org/abs/2105.09938",
+        dataset={
+            "path": "CoIR-Retrieval/apps",
+            "revision": "main",
+        },
+        type="Retrieval",
+        category="p2p",
+        modalities=["text"],
+        eval_splits=[_EVAL_SPLIT],
+        eval_langs=["eng-Latn", "python-Code"],
+        main_score="ndcg_at_10",
+        date=("2021-05-20", "2021-05-20"),
+        domains=["Programming", "Written"],
+        task_subtypes=["Code retrieval"],
+        license="MIT",
+        annotations_creators="derived",
+        dialect=[],
+        sample_creation="found",
+        bibtex_citation="""@article{hendrycksapps2021,
+          title={Measuring Coding Challenge Competence With APPS},
+          author={Dan Hendrycks and Steven Basart and Saurav Kadavath and Mantas Mazeika and Akul Arora and Ethan Guo and Collin Burns and Samir Puranik and Horace He and Dawn Song and Jacob Steinhardt},
+          journal={NeurIPS},
+          year={2021}
+        }""",
+        descriptive_stats={
+            "n_samples": {
+                _EVAL_SPLIT: 1000,
+            },
+        },
```

Comment:
Seems like we need to calculate descriptive stats here. (see e.g.: https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/Retrieval/kor/KoStrategyQA.py) 

You can use the associated function for this: https://github.com/embeddings-benchmark/mteb/blob/main/mteb/abstasks/AbsTaskRetrieval.py#L389

Reply:
We sadly haven't added it to all datasets yet, but once added, we should definitely. do that